gemsfx/src/main/java/com/dlsc/gemsfx/util/SessionManager.java

```diff
@@ -143,12 +156,62 @@ public void register(String path, BooleanProperty property) {
     public void register(String path, StringProperty property) {
         LOG.fine("registering string property at path " + path);
         property.set(preferences.get(path, property.get()));
-        property.addListener((it, oldValue, newValue) -> {
+        initializePreferences(path, property);
+        var listener = (ChangeListener<String>) (it, oldValue, newValue) -> {
             if (newValue != null) {
                 preferences.put(path, newValue);
             } else {
                 preferences.remove(path);
             }
-        });
+        };
+        addListener(property, listener);
+    }
+
+    /**
+     * Unregisters all listeners created by SessionManager for the specified property.
+     *
+     * @param property the property to unregister
+     */
+    public void unregister(Property property) {
+        var listeners = propertyToListeners.get(property);
+        if (listeners != null) {
+            listeners.forEach(listener -> property.removeListener(listener));
+            listeners.clear();
+            propertyToListeners.remove(property);
+        }
+    }
+
+    /**
+     * Unregisters all listeners created by SessionManager.
+     *
+     */
+    public void unregisterAll() {
+        for (var entry : propertyToListeners.entrySet()) {
+            entry.getValue().forEach(listener -> entry.getKey().removeListener(listener));
+            entry.getValue().clear();
+        }
+        propertyToListeners.clear();
+    }
+
+    private void addListener(Property property, ChangeListener listener) {
+        property.addListener(listener);
+        propertyToListeners.computeIfAbsent(property, k -> new ArrayList<>()).add(listener);
+    }
+
+    private void initializePreferences(String path, Property property) {
```

Comment:
Not sure I like the name. Maybe "putPropertyValueIntoPreferences"

Reply:
Fixed.